x-pack/plugins/security_solution/public/common/components/response_actions/endpoint_action_results.tsx

```diff
@@ -0,0 +1,112 @@
+import { ActionsLogExpandedTray } from '../../../management/components/endpoint_response_actions_list/components/action_log_expanded_tray';
+import type { ResponseActionsApiCommandNames } from '../../../../common/endpoint/service/response_actions/constants';
+import { ENDPOINT_COMMANDS } from '../event_details/translations';
+import { ResponseActionsEmptyPrompt } from './response_actions_empty_prompt';
+
+interface EndpointResponseActionResultsProps {
+  action: LogsEndpointAction;
+}
+
+export const EndpointResponseActionResults = ({ action }: EndpointResponseActionResultsProps) => {
+  const { rule, agent } = action;
+  const { action_id: actionId, expiration } = action.EndpointActions;
+  const {
+    endpointPrivileges: { canReadActionsLogManagement, canAccessEndpointActionsLogManagement },
+  } = useUserPrivileges();
+
+  const canReadEndpoint = canReadActionsLogManagement && canAccessEndpointActionsLogManagement;
+  const { data: responseData } = useGetAutomatedActionResponseList(
+    { actionId, expiration, agent },
+    { skip: !canReadEndpoint }
+  );
+
+  const eventText = getCommentText(action.EndpointActions.data.command);
+  const expandedAction = combineAction(action, responseData);
+
+  return (
+    <EuiComment
+      username={rule?.name}
+      timestamp={<FormattedRelative value={action['@timestamp']} />}
+      event={eventText}
+      data-test-subj={'endpoint-results-comment'}
+    >
+      {canReadEndpoint ? (
+        <ActionsLogExpandedTray action={expandedAction} />
+      ) : (
+        <ResponseActionsEmptyPrompt type="endpoint" />
+      )}
+    </EuiComment>
+  );
+};
+
+const getCommentText = (command: ResponseActionsApiCommandNames) => {
+  if (command === 'isolate') {
+    return ENDPOINT_COMMANDS.isolated;
+  }
+  if (command === 'unisolate') {
+    return ENDPOINT_COMMANDS.released;
+  }
+  return `executed command ${command}`;
+};
+
+const combineAction = (
+  action: LogsEndpointAction,
+  responseData: ReturnType<typeof useGetAutomatedActionResponseList>['data']
+) => {
+  const { rule } = action;
+  const { parameters, alert_id: alertId, comment, command } = action.EndpointActions.data;
+
+  const status = responseData?.isExpired
+    ? 'failed'
+    : responseData?.isCompleted
+    ? responseData?.wasSuccessful
+      ? 'successful'
+      : 'failed'
+    : 'pending';
+
+  return {
+    id: action.EndpointActions.action_id,
+    agents: action.agent.id as string[],
+    parameters,
+    ...(alertId?.length ? { alertIds: alertId } : {}),
+    ...(rule
+      ? {
+          ruleId: rule.id,
+          ruleName: rule.name,
+        }
+      : {}),
+    createdBy: action.rule?.name || 'unknown',
+    comment,
+    command,
+    hosts: (action.agent.id as string[]).reduce(
+      (acc: Record<string, { name: string }>, agentId: string) => {
+        acc[agentId] = {
+          name: '',
```

Comment:
shouldn't we display something here?

Reply:
It depends. So far we do not make any use of agent name. However we might fetch agent details the same way endpoint history log is doing.

Reply:
I changed it, and now we do receive the `hosts`